File: shared/src/commonMain/kotlin/com/intive/aifirst/petspot/domain/models/AnimalGender.kt
package com.intive.aifirst.petspot.domain.models

import kotlin.js.ExperimentalJsExport
import kotlin.js.JsExport

/**
 * Animal gender/sex.
 * Displayed as icon on web version (per Figma spec).
 */
@OptIn(ExperimentalJsExport::class)
@JsExport
enum class AnimalGender(val displayName: String) {
    MALE("Male"),
    FEMALE("Female"),
    UNKNOWN("Unknown"),
}

File: iosApp/iosApp/Domain/Models/AnimalGender.swift
import Foundation

/// Animal gender/sex.
enum AnimalGender {
    case male
    case female
    case unknown
    
    /// Failable initializer from string (case-insensitive)
    init?(fromString string: String) {
        let uppercased = string.uppercased()
        switch uppercased {
        case "MALE":
            self = .male
        case "FEMALE":
            self = .female
        case "UNKNOWN":
            self = .unknown
        default:
            return nil
        }
    }
}

